src/Controller/JobsController.php

```diff
@@ -0,0 +1,285 @@
+     * @return Response
+     */
+    public function getJobById(int $id): Response
+    {
+        try {
+            $job = $this->jobsRepository->find($id);
+
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'jobs' => [
+                            'id' => $job->getId(),
+                            'name' => $job->getName(),
+                            'priority' => $job->getPriority(),
+                            'active' => $job->getActive(),
+                            'description' => $job->getDescription(),
+                            'company' => [
+                                $job->getCompany()->getId(),
+                                'name' => $job->getCompany()->getName()
+                            ]
+                        ],
+                        'error' => false,
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'result' => [
+                        'results' => [
+                            'error' => true,
+                            'message' => $e->getMessage()
+                        ]
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updateJobs
+     *
+     * @param  mixed $request
+     * @param  mixed $id
+     * @return Response
+     */
+    public function updateJob(Request $request, int $id): Response
+    {
+        try {
+            $requestParams = $request->query->all();
+
+            $updateResult = $this->jobsRepository->update($id, $requestParams);
+
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'updated' => $updateResult,
+                        'error' => false,
+                    ]
+                ]
+            );
+        } catch (Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * getJobList
+     *
+     * @return Response
+     */
+    public function getJobList(): Response
+    {
+        $jobs = $this->jobsRepository->findAll();
+
+        $name = [];
+
+        foreach ($jobs as $job) {
+            $name[] = $job->getName();
+        }
+
+        return new JsonResponse(
+            [
+                'results' => [
+                    'jobs' => [
+                        'name' => $name,
+                    ],
+                    'error' => false,
+                ]
+            ]
+        );
+    }
+
+    /**
+     * deleteJob
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteJob(int $id): Response
+    {
+        $jobs = $this->jobsRepository->findAll();
```

Comment:
We do not need to get all the list of jobs when we can get exactly the job we need:
Lines 202-206 can be replaced with

` $job = $this->jobsRepository->find($id);
            if (is_null($job)) {
                throw new \InvalidArgumentException('Could not find job with id: ' . $id);
            }   
            $this->jobsRepository->remove($job);`